pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -341,3 +376,182 @@ func validateAWS(h *validatorHandler, m *Machine) (bool, utilerrors.Aggregate) {
 
 	return true, nil
 }
+
+func defaultAzure(h *defaulterHandler, m *Machine) (bool, utilerrors.Aggregate) {
+	klog.V(3).Infof("Defaulting Azure providerSpec")
+
+	var errs []error
+	providerSpec := new(azure.AzureMachineProviderSpec)
+	if err := yaml.Unmarshal(m.Spec.ProviderSpec.Value.Raw, &providerSpec); err != nil {
```

Comment:
Not a big deal, but it might be better to maintain a single location for unmarshaling spec in here: https://github.com/openshift/cluster-api-provider-azure/blob/9034812d3a88fae165bdc359d2c0a04c43f61c7a/pkg/cloud/azure/actuators/machine_scope.go#L218 so it could be removed later, when the  providers will have it's own resources.